tss-esapi-sys/build.rs

```diff
@@ -59,19 +69,158 @@ pub mod tpm2_tss {
     /// The installed tpm2-tss libraries that are of
     /// interest.
     pub struct Installation {
-        _tss2_sys: Library,
+        tss2_sys: Library,
         tss2_esys: Library,
         tss2_tctildr: Library,
         tss2_mu: Library,
         tss2_tcti_tbs: Option<Library>,
     }
 
     impl Installation {
+        fn platform_args() -> Option<Vec<String>> {
+            cfg_if::cfg_if! {
+                if #[cfg(windows)] {
+                    let mut clang_args: Vec<String> = Vec::new();
+                    let hklm = winreg::RegKey::predef(winreg::enums::HKEY_LOCAL_MACHINE);
+                    let sdk_entry = hklm.open_subkey("SOFTWARE\\WOW6432Node\\Microsoft\\Microsoft SDKs\\Windows\\v10.0").unwrap();
+                    let installation_path: String = sdk_entry.get_value("InstallationFolder").unwrap();
+                    let ip_pb = PathBuf::from(installation_path).join("Include");
+                    let windows_sdk = ip_pb.join("10.0.17134.0");
+                    clang_args.push(format!("-I{}", windows_sdk.join("ucrt").display()));
+                    clang_args.push(format!("-I{}", windows_sdk.join("um").display()));
+                    clang_args.push(format!("-I{}", windows_sdk.join("shared").display()));
+                    return Some(clang_args);
+                }
+                else {
+                    return None;
+                }
+            }
+        }
+
+        #[cfg(feature = "bundled")]
+        /// Fetch the given source repo using git
+        fn fetch_source(
+            dest_path: impl AsRef<std::path::Path>,
+            name: &str,
+            repo: &str,
+            branch: &str,
+        ) -> std::path::PathBuf {
+            let parent_path = dest_path.as_ref();
+            let repo_path = parent_path.join(name);
+            if !repo_path.join("Makefile.am").exists() {
+                let output = std::process::Command::new("git")
+                    .args(["clone", repo, "--depth", "1", "--branch", branch])
+                    .current_dir(parent_path)
+                    .output()
+                    .expect(&format!("git clone for {} failed", name));
+                let status = output.status;
+                if !status.success() {
+                    panic!(
+                        "git clone for {} returned failure status {}:\n{:?}",
+                        name, status, output
+                    );
+                }
+            }
+
+            repo_path
+        }
+
+        #[cfg(feature = "bundled")]
+        fn compile_with_autotools(p: PathBuf) -> PathBuf {
+            let output1 = std::process::Command::new("./bootstrap")
+                .current_dir(&p)
+                .output()
+                .expect("bootstrap script failed");
+            let status = output1.status;
+            assert!(
+                status.success(),
+                "bootstrap script failed with {status}:\n{output1:?}"
```

Comment:
The old MSRV tests are run with an old compiler that probably does not support 2021 edition yet so this wont pass that check.